src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php

```diff
@@ -33,8 +34,10 @@
  */
 final class TranslationDefaultDomainNodeVisitor implements NodeVisitorInterface
 {
+    private const INTERNAL_VAR_NAME = '__internal_trans_default_domain';
+
     private Scope $scope;
-    private int $nestingLevel = 0;
+    private int $nameCounter = 0;
```

Comment:
Having a global name counter will not produce reproducible builds (as it depends on the order of compilation of templates). And it does not even prevent conflicts in case the parent template was already cached from a previous compilation (and so not taking into account by the current counter).
There's a good reason why I previously suggested hashing the template name.

Reply:
I changed the implementation back to what I initially had in #58706. But my question from https://github.com/symfony/symfony/pull/58706#discussion_r1820544306 still stands. How do we handle cases where there is no template name? Is using the counter (as I did for now) the way to go?

Reply:
AFAICT, this case cannot happen in a node visitor for traversed nodes. The parser sets the source context in the node during parsing, which sets it recursively in the whole tree. And from that point, setting a child node in a node of the tree will inject the source context in the child node.

Reply:
What would you suggest? Remove the counter and do nothing? Or throw a runtime exception when that happens?

Reply:
okay, I have updated the code accordingly